src/sentry/utils/appleconnect/itunes_connect.py

```diff
@@ -349,3 +320,396 @@ def get_dsym_url(
+
+        :raises: :class:`InvalidTwoFactorAuthError`
+        """
+        assert self.state is ClientState.AUTH_REQUESTED
+        url = "https://idmsa.apple.com/appleauth/auth/verify/trusteddevice/securitycode"
+        logger.debug(f"POST {url}")
+        response = self.session.post(
+            url,
+            json={
+                "securityCode": {
+                    "code": code,
+                }
+            },
+            headers={
+                "scnt": self._scnt,
+                "X-Apple-Id-Session-Id": self._session_id,
+                "Accept": "application/json",
+                "X-Apple-Widget-Key": self.service_key,
+            },
+        )
+        if not response.ok:
+            raise InvalidTwoFactorAuthError
+        else:
+            self.state = ClientState.AUTHENTICATED
+
+    def _request_trusted_phone_info(self) -> None:
+        """Requests the trusted phone info for the account."""
+        assert self.state is ClientState.AUTH_REQUESTED
+        url = "https://idmsa.apple.com/appleauth/auth"
+        logger.debug("GET %s", url)
+        response = self.session.get(
+            url,
+            headers={
+                "scnt": self._scnt,
+                "X-Apple-Id-Session-Id": self._session_id,
+                "Accept": "application/json",
+                "X-Apple-Widget-Key": self.service_key,
+            },
+        )
+        if response.status_code != HTTPStatus.OK:
+            raise ITunesError(f"Unexpected response status: {response.status_code}")
+        info = response.json()["trustedPhoneNumber"]
+        self._trusted_phone = TrustedPhoneInfo(
+            id=info["id"],
+            push_mode=info["pushMode"],
+            number_with_dial_code=info["numberWithDialCode"],
+            obfuscated_number=info["obfuscatedNumber"],
+        )
+
+    def request_sms_auth(self) -> None:
+        """Requests sending the authentication code to a trusted phone.
+
+        :raises ITunesError: if there was an error requesting to use the trusted phone.
+        """
+        assert self.state is ClientState.AUTH_REQUESTED
+        self._request_trusted_phone_info()
+        assert self._trusted_phone is not None
+        url = "https://idmsa.apple.com/appleauth/auth/verify/phone"
+        logger.debug("PUT %s", url)
+        response = self.session.put(
+            url,
+            json={
+                "phoneNumber": {"id": self._trusted_phone.id},
+                "mode": self._trusted_phone.push_mode,
+            },
+            headers={
+                "scnt": self._scnt,
+                "X-Apple-Id-Session-Id": self._session_id,
+                "Accept": "application/json",
+                "X-Apple-Widget-Key": self.service_key,
+                "Content-Type": "application/json",
+            },
+        )
+        if response.status_code != HTTPStatus.OK:
+            raise ITunesError("Unexpected response status: {response.status_code}")
```

Comment:
```suggestion
            raise ITunesError(f"Unexpected response status: {response.status_code}")
```

otherwise this would be a literal `{response.status_code}`, right?

Reply:
right, good catch